src/main/scala/advancepassengerinfo/importer/provider/FileNames.scala

```diff
@@ -8,12 +9,23 @@ import scala.jdk.CollectionConverters.CollectionHasAsScala
 import scala.jdk.FutureConverters.CompletionStageOps
 
 trait FileNames {
-  val nextFiles: String => Future[List[String]]
+  val nextFiles: (String, String) => Future[List[String]]
+
+  val s3Files: String => Future[List[String]]
 }
 
 case class S3FileNames(s3Client: S3AsyncClient, bucket: String)
                       (implicit ec: ExecutionContext) extends FileNames {
-  override val nextFiles: String => Future[List[String]] = (lastFile: String) => s3Client
+  private val log = Logger(getClass)
+
+  val s3Files: String => Future[List[String]] = lastFile => s3Client
     .listObjects(ListObjectsRequest.builder().bucket(bucket).marker(lastFile).build()).asScala
     .map(_.contents().asScala.map(_.key()).toList)
+
+  override val nextFiles: (String, String) => Future[List[String]] = (lastFile: String, fallbackFileName) => s3Files(lastFile)
```

Comment:
When we fail to fetch the file names, rather than returning a made up file name we just need to return an empty list. That's the same as there not yet being any more files. So the app will try again in a minute to fetch the list of files, and hopefully if it was a temporary network issue then it will succeed the second time around

Reply:
Done